DSharpPlus/Logging/ShardedLoggerFactory.cs

```diff
@@ -26,23 +26,18 @@
 
 namespace DSharpPlus
 {
-    internal class ShardedLoggerFactory : ILoggerFactory
+    internal class ShardedLoggerFactory<T> : ILoggerFactory
```

Comment:
Consider `where T : BaseDiscordClient`, as that seems to be the only appropriate use of this class.

Reply:
Sharded nor Clustered client extend the BaseDiscordClient, though